Change AfterShip polling interval from 5m to 15m

dudyn5ky1 opened the pull request:
AfterShip API free tier only allows 100 requests per day, with current refresh interval (each 5 minutes) we can exceed the limit in ~8 hours. This PR introduces simple solution for this - change refresh interval from 5 to 15 minutes. Issue: #42356

<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
I'm changing refresh interval for AfterShip from 5 to 15 minutes. This way users of the free-tier won't exceed API request limit (100 requests / day).

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example entry for `configuration.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

```yaml
# Example configuration.yaml
sensor:
  - platform: aftership
    api_key: <API_KEY>

```

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #42356
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

homeassistant commented:
Hi @dudyn5ky1, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#42360).

Once you do that we will be able to review and accept this pull request.

Thanks!

dudyn5ky1 commented:
Signed CLA.

dudyn5ky1 commented:
Still fixing one issue

frenck commented on homeassistant/components/aftership/sensor.py:
```diff
@@ -52,6 +52,7 @@
     {
         vol.Required(CONF_API_KEY): cv.string,
         vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_REFRESH_TIME, default=DEFAULT_REFRESH_TIME): cv.positive_int,
```
We no longer accept YAML configuration for integrations that connect to devices or services.

Instead, a configuration flow via the UI should be used.

For more information about this decision see:

- ADR-0007: <https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md#decision>
- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

To get started on the integration configuration, see our developer documentation:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

dudyn5ky1 replied:
Thanks for the comment. 

Such change will take some time, how about I change default interval from 5 to 15 (without yaml config change) now and do the migration to UI flow in the other one?

5 to 15 change will fix the API request limit error for the free users of AfterShip, since now the widget doesn't work.

dudyn5ky1 committed:
[AfterShip] Change refresh interval from 5 to 15

This change will prevent reaching API requests limit of AfterShip free-tier account.

dudyn5ky1 commented:
Hey @frenck,

I've removed all configuration changes, now this PR is only about changing interval from 5 to 15 minutes for AfterShip. This way we avoid exceeding API request limit.

frenck reviewed (approved):
That is also a solution 👍 

Thanks, @dudyn5ky1 🎉
